pkg/resourcetopologyexporter/resourcetopologyexporter.go

```diff
@@ -173,8 +200,10 @@ func (rm *ResourceMonitor) Run(eventsChan <-chan PollTrigger) (<-chan nrtupdater
 
 				if err != nil {
 					klog.Warningf("failed to scan pod resources: %w\n", err)
+					rm.UpdateCondition(v1.ConditionFalse)
 					continue
 				}
+				rm.UpdateCondition(v1.ConditionTrue)
```

Comment:
I'm not sure we should do this synchronously. I'd be happier if we update the condition asynchronously in a goroutine

Reply:
Sure